### PR TITLE
Week 4 scott tuschl

### PR DESCRIPTION
**Pull request by scott198989**
# Submitting Week 4 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -13,10 +13,41 @@
...
 
 // a) Create a test with an expect statement using the variable provided.
+
+describe('stFunct1',  () => {
+  it('takes in an array and removes the first item shuffling the remaining content', () => {
+    expect(stFunct1(colors1)).toEqual(expect.arrayContaining(['blue', 'green', 'yellow', 'pink']))
+  })
+})
+
+describe('stFunct2', () => {
```
Comment (chrisyhlee) [inline code-challenges.test.js:23]: Remember what I've been saying in assessment reviews and intros, you don't need separate tests for each variable.

File: code-challenges.test.js [hunk 2]
```diff
@@ -13,10 +13,41 @@
...
+
+//   ReferenceError: stFunct1 is not defined
+// FAIL  ./code-challenges.test.js
+// ● Test suite failed to run
+
+//   ReferenceError: stFunct2 is not defined
+
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
+//Psudeocode: The function takes an array as input, then it creates a new array called newArray by extracting elements from colors1 starting from index 1 (excluding the first element). Then, it shuffles the elements in newArray using the .sort() method with a random comparison function (() => 0.5 - Math.random()). The stFunct function is called with colors1 as an argument, and the resulting shuffled array is stored in the finalArray variable. The finalArray is then displayed or printed to the console. The process is the same exact way for both array of colors (colors1, colors2), all that was changed was naming conventions.
```
Comment (chrisyhlee) [inline code-challenges.test.js:40]: As you said in the pseudo code, the process is the same for each variable, as it should be. This means there's no need to declare 2 separate functions for the 2 variables. Also, no need to store your function call in a variable like you did on line 47.

File: code-challenges.test.js [hunk 3]
```diff
@@ -25,27 +56,148 @@ const colors2 = [
...
 const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+
+describe('combine', () => {
+  it('combines two arrays and removes duplicates', () => {
+    expect(combine(["array", "object", "number"], ["string", "Boolean", "null", "undefined"])).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
+
+
+const combine = (array1, array2) => {
```
Comment (chrisyhlee) [inline code-challenges.test.js:186]: I encourage you to refactor this without any variables being created!

**Review by chrisyhlee — APPROVED** [review]
Scott, you are definitely progressing in your pseudo code and logic each week. The biggest thing I noticed is that you are making separate tests and functions/methods for every variable that the prompt is giving you. The whole point of functions/methods is that they should be dynamic. This means you should be able to pass in every test variable into a single function. Because of this, your assessment displays Novice level. However, get away from this bad habit and you will be at Advanced Beginner! Keep up the great work

**Comment by scott198989** [comment]
NA

**Pull request merged**